sim/firesim-lib/src/main/scala/bridges/DromajoBridge.scala

```diff
@@ -0,0 +1,178 @@
+            Cat(boolPad(trace.exception, 8),
+              Cat(trace.priv.asUInt.pad(8),
+                Cat(trace.wdata.pad(wdataWidth),
+                  Cat(trace.insn.pad(insnWidth),
+                    Cat(trace.iaddr.pad(iaddrWidth), boolPad(trace.valid, 8)))))))))
+    }
+
+    val maxTraceSize = paddedTraces.map(t => t.getWidth).max
+    val outDataSzBits = outgoingPCISdat.io.enq.bits.getWidth
+
+    // constant
+    val totalStreamsPerToken = 2 // minTraceSz==190b so round up to nearest is 256b
+                                 // constant
+
+    require(maxTraceSize < (outDataSzBits / totalStreamsPerToken), "All instruction trace bits must fit in 256b")
+
+    // how many streams being sent over
+    val numStreams = traces.size
+    // num tokens needed to display full stream
+    val numTokenForAll = ((numStreams - 1) / totalStreamsPerToken) + 1
+
+    // tracequeue is full
+    val traceQueueFull = outgoingPCISdat.io.deq.valid && !outgoingPCISdat.io.enq.ready
+
+    // only inc the counter when the something is sent... input is valid... and output is avail
+    val counterFire = outgoingPCISdat.io.enq.fire() && hPort.toHost.hValid && !traceQueueFull
+    val (cnt, wrap) = Counter(counterFire, numTokenForAll)
+
+    val paddedTracesAligned = paddedTraces.map(t => t.asUInt.pad(outDataSzBits/totalStreamsPerToken))
+    val paddedTracesTruncated = if (numStreams == 1) {
+      (paddedTracesAligned.asUInt >> (outDataSzBits.U * cnt))
+    } else {
+      (paddedTracesAligned.asUInt >> (outDataSzBits.U * cnt))(outDataSzBits-1, 0)
+    }
+
+    outgoingPCISdat.io.enq.valid := hPort.toHost.hValid
+    outgoingPCISdat.io.enq.bits := paddedTracesTruncated
+
+    // tell the host that you are ready to get more
+    hPort.toHost.hReady := outgoingPCISdat.io.enq.ready && wrap
+
+    // This is uni-directional. We don't drive tokens back to the target.
+    hPort.fromHost.hValid := true.B
+
+    // tell c driver that queue is full using MMIO
+    attach(traceQueueFull, "trace_queue_full", ReadOnly)
+
+    genCRFile()
+
+    // modify the output header file
+    override def genHeader(base: BigInt, sb: StringBuilder) {
+      super.genHeader(base, sb)
+
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_iaddr_width", UInt32(iaddrWidth)))
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_insn_width", UInt32(insnWidth)))
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_wdata_width", UInt32(wdataWidth)))
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_cause_width", UInt32(causeWidth)))
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_tval_width", UInt32(tvalWidth)))
+      sb.append(CppGenerationUtils.genMacro(s"${getWName.toUpperCase}_num_streams", UInt32(numStreams)))
+    }
+
+    // general information printout
+    println(s"Dromajo Bridge Information")
+    println(s"  Total Inst. Streams: ${numStreams}")
```

Comment:
As I mentioned previously I would really prefer that we be consistent in our nomenclature regarding "streams" or machine "width" across tracerV and Fromajo. 

I'm pretty use in the RISC-V universe we use stream =~ the sequence of instructions fetched from one PC.

Reply:
I renamed things to traces as that seems to be what the common terminology is.